utl/dump_options/CMakeLists.txt

```diff
@@ -1,2 +1,2 @@
 add_executable(vw-dump-options main.cc)
-target_link_libraries(vw-dump-options vw RapidJSON)
+target_link_libraries(vw-dump-options PUBLIC VowpalWabbit::vw RapidJSON)
```

Comment:
Is there a reason this is PUBLIC?

Reply:
You should always specify PUBLIC/PRIVATE/INTERFACE. This is an exe so it doesn't really matter much so I defaulted to PUBLIC